proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,140 @@
+# dotnet nuget why command
+
+- Status: **Draft**
+- Author: [Kartheek Penagamuri](https://github.com/kartheekp-ms)
+- GitHub Issue [11782](https://github.com/NuGet/Home/issues/11782)
+
+## Problem Background
+
+Developers are often confused with top-level & transitive dependencies in their project. A developer should be able understand where every package originated from. A user should be able to promote a transitive dependency quickly & easily to a top-level dependency in the case there is a security concern on the package version depended on or if promoting to a top-level package will resolve a conflict.
+
+Knowing how a package is being included in a project helps a developer diagnose many problems. It also provides them with an opportunity to understand complex dependency chains they may have within their projects. There is not a great solution that exists to understand the nature of top-level packages and their transitive dependencies. The solution explorer in Visual Studio does the best job at allowing a user to dive into each NuGet package & see all transitive dependencies from the top-level dependency.
+
+![](../../meta/resources/TransitiveDependencies/SolutionView.png)
+
+The dotnet CLI does not provide the insight into “why” a transitive dependency is listed, although it does list everything that has been resolved.
```

Comment:
Maybe instead of `why` could `explain` would be better word?  For example [sql explain](https://www.geeksforgeeks.org/explain-in-sql/) does similar  action for sql.

Reply:
Isn't `why` play on the command name here?

Reply:
If explain is already used in sql, I personally think it would be best to follow the same naming convention here too. Any other suggestions/things to consider?

Reply:
The `why` name comes from npm's why command, which is the exact same problem space as the one we're solving.

Reply:
Should try to remain consistent with other package managers such as 'why' or 'tree'. SQL explain is a good prior art though

Reply:
Ok, I am convinced. Also based on this https://github.com/NuGet/Home/pull/11875#discussion_r897458421, it seems like @kartheekp-ms and @chgill-MSFT agree that `dotnet nuget why` would be best.